ProjFS.Mac/PrjFSKext/KauthHandler.cpp

```diff
@@ -433,6 +404,36 @@ KEXT_STATIC int HandleVnodeOperation(
         }
     }
     
+    if (isDeleteAction)
+    {
+        // Allow any user to delete individual files, as this generally doesn't cause nested kauth callbacks.
+        ProviderCallbackPolicy callbackPolicy = isDirectory ? CallbackPolicy_UserInitiatedOnly : CallbackPolicy_AllowAny;
+        if (!TryGetVirtualizationRoot(&perfTracer, context, currentVnode, pid, callbackPolicy, &root, &vnodeFsidInode, &kauthResult, kauthError))
+        {
+            goto CleanupAndReturn;
+        }
+                
+        PerfSample preDeleteSample(&perfTracer, PrjFSPerfCounter_VnodeOp_PreDelete);
```

Comment:
Could you add a comment explaining why the PreDeletes are sent after the hydrate messages?